### PR TITLE
Fix Burn bindings mismatch

### LINKED ISSUE #1607: [Bug]: Typescript Burn Interface mismatch
### Issue description

The [Typescript Burn Interface](https://github.com/iotaledger/iota.rs/blob/develop/client/bindings/nodejs/types/burn.ts) seems to be misaligned. I guess all keys should be optional and `nativeTokens` should be an optional array as well.

### Version

Wasm `1.0.0-alpha.1` and latest develop.

### Expected behaviour

Passing Aliases, NFTs, Foundries and native Tokens is optional. It is possible to pass multiple native Tokens.

### Actual behaviour

Existence of all keys is enforced and only a single native Token is considered valid.

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue
 
```
this.client.buildAndPostBlock(secretManager, {
    inputs: [aliasInput],
    outputs: [basicOutput],
    burn: {
        aliases: []
    }
});
```

### Errors

```
Type '{ aliases: never[]; }' is missing the following properties from type 'Burn': nfts, foundries, nativeTokens
Type 'never[]' is missing the following properties from type 'INativeToken': id, amount
```

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fix https://github.com/iotaledger/iota.rs/issues/1607

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/api/block_builder/input_selection/core/burn.rs [hunk 1]
```diff
@@ -111,28 +111,32 @@ impl Burn {
...
     fn from(value: &Burn) -> Self {
         Self {
-            aliases: value.aliases.clone(),
-            nfts: value.nfts.clone(),
-            foundries: value.foundries.clone(),
-            native_tokens: HashMap::from_iter(
+            aliases: Some(value.aliases.clone()),
+            nfts: Some(value.nfts.clone()),
+            foundries: Some(value.foundries.clone()),
+            native_tokens: Some(HashMap::from_iter(
```
Comment (Thoralf-M) [inline client/src/api/block_builder/input_selection/core/burn.rs]: Shouldn't we check if they're empty and put `None` then?

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/burn.rs]: Done

File: client/src/api/block_builder/input_selection/core/burn.rs [hunk 2]
```diff
@@ -111,28 +111,48 @@ impl Burn {
...
-                value
-                    .native_tokens
-                    .iter()
-                    .map(|(token_id, amount)| (*token_id, U256Dto::from(amount))),
-            ),
+            aliases: if value.aliases.is_empty() {
+                None
+            } else {
+                Some(value.aliases.clone())
+            },
```
Comment (Thoralf-M) [inline client/src/api/block_builder/input_selection/core/burn.rs]: Does this work and would you prefer that?
```suggestion
            aliases: value.aliases.is_empty().then_some(value.aliases.clone()),
```

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/burn.rs]: But that would do `Some` when it's empty

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/burn.rs]: Unfortunately there is no `is_full` or whatever

Comment (Thoralf-M) [reply client/src/api/block_builder/input_selection/core/burn.rs]: Ah yeah, but what about `(!value.aliases.is_empty())`?

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/burn.rs]: Should work, I didn't think of that

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/burn.rs]: Don't think I ever used then_some before, good stuff

Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/burn.rs]: Done

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Please also add an entry to the wasm changelog, since it will have the same changes as the nodejs binding

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**